### ISSUE TITLE
Timing tests

### ISSUE THREAD

**Issue by joelberkeley** [stefan-hoeck/idris2-hedgehog#44]
I'd like to time the tests, but the `test` function exits the program so I can't stick a `System.time` before and after tests

**Comment by buzden** [comment]
Do you want time being printed for a property or, say, min/max/av of each test in each property?

Or do you want to measure time just once? In this case, you can run `checkGroup` instead of `test`?

**Comment by joelberkeley** [comment]
Detailed timing hadn't occured to me, but that would be cool. I imagine per property or group would be ideal. Thanks for the tip re `checkGroup`

**Comment by joelberkeley** [comment]
I guess I could try to contribute this function myself. Would `System.time` be the best? Millisecond timing would be preferrable

**Comment by joelberkeley** [comment]
it would require adding IO to the runner. Is that a good idea?

**Comment by buzden** [comment]
> it would require adding IO to the runner. Is that a good idea?

No, it's not. Everything is abstracted in the runner to be able to run Hedgehog in different environment, to test Hedgehog by Hedgehog in particular. So, every functionality is moved to a separate interface, like it's done with `HasTerminal` and `CanInitSeed` and similar interfaces.

I personally think there is no need to invent your own interface for this, since there is a library with such interface already: https://github.com/buzden/idris2-time-for-sleep/blob/master/src/System/Time.idr#L151
@stefan-hoeck recently said that he does not mind adding external dependencies to well-maintained well-document libraries. If currently this library is not considered so, well, I can put some effort to document it. Also, this lacks some `Interpolation` instances for time types to have nicely printable time, but I'll do this, I wanted to do this anyway.

**Comment by joelberkeley** [comment]
this is more complicated than I expected. I don't have the head space for something non-trivial, sorry

**Comment by joelberkeley** [comment]
but yes I see the pattern and I see how your lib fits that pattern
